test/Common/HttpApiHandlerTests.cs

```diff
@@ -807,6 +807,40 @@ public async Task GetQueryWithoutRuntimeStatus_is_Success()
             Assert.Equal("Completed", actual[1].RuntimeStatus);
         }
 
+        [Fact]
+        [Trait("Category", PlatformSpecificHelpers.TestCategory)]
+        public async Task HandleGetStatusRequestAsync_Correctly_Parses_InstanceId_With_Spaces()
+        {
+            var list = (IList<DurableOrchestrationStatus>)new List<DurableOrchestrationStatus>
+            {
+                new DurableOrchestrationStatus
+                {
+                    Name = "DoThis",
+                    InstanceId = "01",
+                    RuntimeStatus = OrchestrationRuntimeStatus.Completed,
+                },
+            };
+
+            var instanceId = "test instance id with spaces";
+            var clientMock = new Mock<IDurableClient>();
+            clientMock
+                .Setup(x => x.GetStatusAsync(instanceId, false, false, true))
+                .Returns(Task.FromResult(list.First()));
+            var httpApiHandler = new ExtendedHttpApiHandler(clientMock.Object);
+
+            var getStatusRequestUriBuilder = new UriBuilder(TestConstants.NotificationUrl);
+            getStatusRequestUriBuilder.Path += $"/Instances/" + WebUtility.UrlEncode(instanceId);
+
+            var responseMessage = await httpApiHandler.HandleRequestAsync(
+                new HttpRequestMessage
+                {
+                    Method = HttpMethod.Get,
+                    RequestUri = getStatusRequestUriBuilder.Uri,
+                });
+
+            Assert.Equal(HttpStatusCode.OK, responseMessage.StatusCode);

```

Comment:
Can we also add a assert here to make sure the instanceId returned is correct?

Reply:
Done :)